### PR TITLE
create symlink to last test run output for local profiles

### PR DESCRIPTION
**Pull request by clarkb7**
### What does this PR do?

create a symlink to the last test run output for local profiles

```
$ tree -d -L 1 ~/e2e-output
/home/brandenclark/e2e-output
├── 2024-01-31_15-25-27_3733236386
├── 2024-01-31_21-19-38_1584804161
├── 2024-01-31_21-28-59_3766096043
└── latest -> /home/brandenclark/e2e-output/2024-01-31_21-28-59_3766096043
```

### Motivation

make it easier to examine results from previous test run

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: test/new-e2e/pkg/runner/local_profile.go [hunk 1]
```diff
@@ -114,3 +115,21 @@ func (p localProfile) NamePrefix() string {
...
+// GetOutputDir extends baseProfile.GetOutputDir to create a symlink to the latest run
+func (p localProfile) GetOutputDir() (string, error) {
+	outDir, err := p.baseProfile.GetOutputDir()
+	if err != nil {
+		return "", err
+	}
+
+	// Create a symlink to the latest run for user convenience
+	latestLink := filepath.Join(filepath.Dir(outDir), "latest")
+	_ = os.Remove(latestLink)
```
Comment (pducolin) [inline test/new-e2e/pkg/runner/local_profile.go:128]: **❓ question**
Can this fail ?

**Review by pducolin — APPROVED** [review]
Nice !

File: test/new-e2e/pkg/runner/local_profile.go (hunk 1 above)
Comment (clarkb7) [reply test/new-e2e/pkg/runner/local_profile.go]: updated thanks!

**Pull request merged**